lib/blocks.php

```diff
@@ -372,3 +372,30 @@ function gutenberg_register_legacy_social_link_blocks() {
 }
 
 add_action( 'init', 'gutenberg_register_legacy_social_link_blocks' );
+
+/**
+ * Migrate the legacy `sync_status` meta key to the new `wp_pattern_sync_status` meta key.
+ *
+ * @since 16.1.1
+ *
+ * @param mixed  $value     The value to return, either a single metadata value or an array of values depending on the value of $single.
+ * @param int    $object_id ID of the object metadata is for.
+ * @param string $meta_key  Metadata key.
+ * @param bool   $single    Whether to return only the first value of the specified $meta_key.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/52232
+ */
+function gutenberg_legacy_wp_block_post_meta( $value, $object_id, $meta_key, $single ) {
+	if ( 'wp_pattern_sync_status' !== $meta_key ) {
+		return $value;
+	}
+
+	$sync_status = get_post_meta( $object_id, 'sync_status', $single );
+
+	if ( isset( $sync_status[0] ) && 'unsynced' === $sync_status[0] ) {
```

Comment:
Just thinking aloud, do we care that `'sync_status'` exists. If we detect the same value could we `delete_post_meta` it?

I guess that runs of the risk of deleting third-party  post meta with the same key

Reply:
We decided migrating the data was too risky (although a minimal risk of being too risky), so just leaving the data in place.